### ISSUE TITLE
Alternate Routes

### ISSUE THREAD

**Issue by starwatcher11** [osmandapp/OsmAnd#11631]
# 🚀 feature request

### Description
It would be useful, time-saving and more user-friendly for OsmAnd to show alternative routes on the map when looking for directions.
I know that it is possible to choose a different route with options (avoid roads, fuel-efficient, avoid highways and so on) but that takes time and it isn't useful when you are on the move and need directions right away.
Not only that but those options don't always suggest the route that I want to follow. Even If I choose to go on my own route, OsmAnd keeps giving me directions for its own default route (telling me to turn back) which I don't want because it doesn't suit me.
In short it would be useful for OsmAnd to show alternative routes on the map when asking for directions. Suggested alternative routes could be ones that 1. avoid highways, 2. least distance, 3. route with least amount of traffic lights, 4. more direct route (less turns).
I have to admit that very often I had to quit OsmAnd and use another app for navigation because the suggested route wasn't always the best one and having to configure OsmAnd to offer alternative routes throught various options each time was time-consuming and a struggle.

Please prioritize this feature, it's not just a "Nice to Have" feature but it's an important one that the lack of it often makes OsmAnd unusable. I've been waiting for this kind of a feature since I first discovered OsmAnd years ago.
I know that this issue already exists (Route alternatives #1294) but it's been 6 years since 2015 when it was created in github but it hasn't been implemented yet and there isn't any hope of seeing it in the foreseeable future.

**Comment by tradavyvy** [comment]
Although every extra feature is possibly nice, I would not call osmand 'unusable' without these alturnative routes. I find it is usualy quite do-able to re-route osmand by adding one or two intermediate points in navigation.
It might also be difficult to 'guess' wich alternatives are usefull for wich specific user.
Besides it is already possible to preset your preference for shortest, fastest or avoid highway settings.

**Comment by starwatcher11** [comment]
Indeed, it would be possible for someone to re-route OsmAnd and make it show the route that suits you, however that takes time and when you're in the driver's seat you're looking for directions right away.
Besides it being time-consuming, wouldn't it be more user-friendly for OsmAnd to suggest alternative routes on the map by default? Many other popular navigation apps do that. I'm not saying that OsmAnd should copy other apps but one of the disadvantages of OsmAnd that it isn't very user friendly for non tech-savvy users and is difficult to recommend it to friends and family.
You would point out that I'm free to use other navigation apps but the point is for OsmAnd to become a worthy or even better alternative navigation app, that's why I'm making this feature request. To be honest, I've been using OsmAnd for 1-2 years now, but I haven't seen any BIG new feature that makes the app more user-friendly and easier to use and faster to get directions when in a hurry, but instead OsmAnd keeps falling behind other navigation apps.
I think that my feature request "Alternate Routes" is a meaningful feature towards making OsmAnd easier, better and more accessible to non tech-savvy users.
As for your argument that it is difficult to guess which alternatives are useful for each user, you could take as example what alternatives other apps show, some of them I already mentioned:
1. avoid highways, 2. least distance, 3. route with least amount of traffic lights, 4. more direct route (less turns), 5. avoid tolls

**Comment by starwatcher11** [comment]
Thank you for adding this feature to the future-backend milestone, at least now it's definite that we'll see it implemented someday

**Comment by thompehw** [comment]
@starwatcher11
> Not only that, but even If I start navigation within the app and I drive a different route, OsmAnd keeps giving me directions for its own route, telling me to turn back, make U-turn etc. thus it becomes UNUSABLE in some situations.

Have you tried turning off: navigation_pane>Options>Navigation_settings>Route parameters>Smart_route_reclaculation ? (It can also be done for a whole profile, instead of only for the current route.) If you do this, then when you deliberately drive another route, OsmAnd will keep recalculating the route all the way to your destination instead of directing you back to the original route. You might also want to adjust "Minimal distance to recalculate route".

You loose navigation while the recalculation is happening, so your device needs to be fast enough to recalculate a route of the required length before you next rely on it for a navigation instruction. In my experience this is difficult to guarantee, so it's right that the default is for "Smart route recalculation" to be enabled.

**Comment by Noitarud** [comment]
Sometimes i wonder why it doesn't route along an apparently shorter path, why did it disqualify the path across a V route for example. Maybe let this info be accessible, undoubtedly the path would have been considered. Perhaps the negotiability increases time or speed limit or there is a climb.

**Comment by vshcherb** [comment]
Since we've introduced faster routing, now we have possibility to calculate 1) alternative routes based on key points 2) just calculate 2-3 routes based on different parameters

**Comment by scaidermern** [comment]
Duplicate of #1294?
